src/Control/CLIRequestBuilder.php

```diff
@@ -66,4 +66,18 @@ public static function cleanEnvironment(array $variables)
         // Parse rest of variables as standard
         return parent::cleanEnvironment($variables);
     }
+
+    /**
+     * @param array $variables
+     * @param string $input
+     * @return HTTPRequest
+     */
+    public static function createFromVariables(array $variables, $input)
+    {
+        $request = parent::createFromVariables($variables, $input);
+        // unset scheme so that SS_BASE_URL can provide `is_https` information if required
+        $request->setScheme(null);
```

Comment:
How about we set the scheme from the `SS_BASE_URL` env var instead?

Reply:
that happens anyway, in a way, through director

Reply:
if we implement our own parser of `SS_BASE_URL` here rather than rely on `Director` then we're going to duplicate code. It'll also mean we still have the same problem where the `is_https` check never actually examines the `SS_BASE_URL` because the request object always has a scheme set.

Reply:
Director doesn't setScheme(). HTTPRequest should be the source of truth and we need to stop relying on static state / helpers such as Director.

The only code duplication is `parse_url(Environment::getEnv('SS_BASE_URL'), PHP_URL_SCHEME)`

Reply:
Director::is_https() doesn't even touch SS_BASE_URL so that's wrong to rely on too. It calls getScheme() on the request which is why you need to set it here.

Reply:
I wouldn't say never, but it's certainly not the primary source of truth.

For cli my idea is that we shift to symfony console and have a separate application class. Cli would not have httprequests at all

Reply:
So, just to make sure I've understood - we want to manually parse `SS_BASE_URL` for scheme here (if it's set)?